scripts/do-release.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2022 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""Release automation script."""
+
+import os
+import re
+import subprocess
+import sys
+from abc import abstractproperty
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import tomli
+from packaging.version import Version
+
+ROOT = Path(__file__).parent.parent
+
+
+class BaseCredentials:
```

Comment:
Do we really need this parent abstract class? if no other uses, just define `EnvCredentials` straightaway.

Reply:
fixed